data-prepper-plugins/otel-metrics-source/src/main/java/com/amazon/dataprepper/plugins/source/otelmetrics/OTelMetricSourceConfig.java

```diff
@@ -0,0 +1,200 @@
+/*
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  The OpenSearch Contributors require contributions made to
+ *  this file be licensed under the Apache-2.0 license or a
+ *  compatible open source license.
+ *
+ *  Modifications Copyright OpenSearch Contributors. See
+ *  GitHub history for details.
+ */
+
+package com.amazon.dataprepper.plugins.source.otelmetrics;
+
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class OTelMetricSourceConfig {
```

Comment:
Data Prepper now supports POJO models for plugin configurations using Jackson serialization and it supports Java bean validation. This isn't a mandatory change since it can be made without a major impact later on. But, I did want to mention it.

You can see how we have it working for Traces now:

https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceSourceConfig.java